pizzaserver-server-api/src/main/java/io/github/pizzaserver/api/block/impl/BlockLeaves.java

```diff
@@ -0,0 +1,109 @@
+package io.github.pizzaserver.api.block.impl;
+
+import com.nukkitx.nbt.NbtMap;
+import io.github.pizzaserver.api.block.Block;
+import io.github.pizzaserver.api.block.BlockID;
+import io.github.pizzaserver.api.block.data.LeaveType;
+import io.github.pizzaserver.api.block.descriptors.Flammable;
+import io.github.pizzaserver.api.entity.Entity;
+import io.github.pizzaserver.api.item.Item;
+import io.github.pizzaserver.api.item.data.ToolType;
+import io.github.pizzaserver.api.item.descriptors.ToolItem;
+
+import java.util.*;
+
+public class BlockLeaves extends Block implements Flammable {
+
+    protected LeaveType leaveType;
```

Comment:
While I'm aware you are in the middle of working on this,
there's no need for this considering that the leavetype can be determined by the block state index.